cmd/kube-apiserver/app/server.go

```diff
@@ -258,8 +258,12 @@ func CreateNodeDialer(s completedServerRunOptions) (tunneler.Tunneler, *http.Tra
 		// Use the nodeTunneler's dialer when proxying to pods, services, and nodes
 		proxyDialerFn = nodeTunneler.Dial
 	}
-	// Proxying to pods and services is IP-based... don't expect to be able to verify the hostname
-	proxyTLSClientConfig := &tls.Config{InsecureSkipVerify: true}
+
+	proxyTLSClientConfig := &tls.Config{}
+	if s.KubeletConfig.CAFile == "" && s.KubeletConfig.CAData == nil {
```

Comment:
this config isn't used to proxy to the kubelet, it's used to proxy to pods and services, and node ports other than the kubelet port

Reply:
the transport building to talk to a kubelet happens here:

https://github.com/kubernetes/kubernetes/blob/65a819338a640faf6ddee9d6eea4559a20106c56/pkg/registry/core/node/storage/storage.go#L122-L134

https://github.com/kubernetes/kubernetes/blob/65a819338a640faf6ddee9d6eea4559a20106c56/pkg/kubelet/client/kubelet_client.go#L130-L136

https://github.com/kubernetes/kubernetes/blob/65a819338a640faf6ddee9d6eea4559a20106c56/pkg/kubelet/client/kubelet_client.go#L68-L102

and is passed to pod storage for use by the exec/attach/log subresource handlers here:
 
https://github.com/kubernetes/kubernetes/blob/65a819338a640faf6ddee9d6eea4559a20106c56/pkg/registry/core/rest/storage_core.go#L135-L140